### PR TITLE
Improve new_group example in the context of SyncBatchNorm 

### LINKED ISSUE #48804: Improve torch.distributed.new_group() documentation in the context of SyncBatchNorm
## 📚 Documentation

In current documentation for using subgroups other than the whole world in SyncBatchNorm: https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html#torch.nn.SyncBatchNorm.convert_sync_batchnorm, it is not clear that when creating subgroups, every rank still needs to make the call regardless of whether they are in the group or not, as per the docs in https://pytorch.org/docs/master/distributed.html#torch.distributed.new_group. 

For example, if we have ranks [0-7] and want to create 2 subgroups and sync BN stats across 2 groups separately, we should do:

```
module = MyModule()
r1 = [0, 1, 2, 3]
r2 = [4, 5, 6, 7]
process_groups = [torch.distributed.new_group(pids) for pids in [r1, r2]]
pg_for_rank = process_groups[0 if self.rank <= 3 else 1]
sync_bn_module = torch.nn.SyncBatchNorm.convert_sync_batchnorm(module, pg_for_rank)
```

Note that in this example, every rank calls into `new_group()` for all groups being created, but we pass in the specific process group for that rank into `convert_sync_batchnorm`. 

cc @jlin27 @mruberry @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd

### PR DESCRIPTION
**Pull request by rohan-varma**
Closes https://github.com/pytorch/pytorch/issues/48804
Improves some documentation/example in SyncBN docs to clearly show that each rank must call into all `new_group()` calls for creating process subgroups, even if they are not going to be part of that particular subgroup.
We then pick the right group, i.e. the group that the rank is part of, and pass that into the SyncBN APIs. 

Doc rendering: 

syncbn_update

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -435,7 +435,13 @@ class SyncBatchNorm(_BatchNorm):
         >>> m = nn.SyncBatchNorm(100)
         >>> # creating process group (optional)
         >>> # process_ids is a list of int identifying rank ids.
-        >>> process_group = torch.distributed.new_group(process_ids)
+        >>> process_ids = list(range(8))
```
Comment (pritamdamania87) [inline torch/nn/modules/batchnorm.py:438]: nit: maybe call these `ranks` instead of process_ids since it might be confused with OS pids.

**Review by pritamdamania87 — APPROVED** [review]

**Comment by rohan-varma** [comment]
Lint failure is unrelated: ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements-flake8.txt'
